keps/sig-cluster-lifecycle/0014-20180707-componentconfig-api-types-to-staging.md

```diff
@@ -0,0 +1,296 @@
+* **External Package with defaulting & conversions**: `k8s.io/apimachinery/pkg/apis/config/v1alpha1/types.go`
+* **Internal Package**: `k8s.io/apimachinery/pkg/apis/config/types.go`
+* Structs to be hosted:
+    * ClientConnectionConfiguration
+* Assignee: ?
+
+#### k8s.io/apiserver changes
+
+* **Not a "real" API group, instead shared packages only with both external and internal types.**
+* **External Package with defaulting & conversions**: `k8s.io/apiserver/pkg/apis/config/v1alpha1/types.go`
+* **Internal Package**: `k8s.io/apiserver/pkg/apis/config/types.go`
+* Structs to be hosted:
+    * LeaderElectionConfiguration
+    * DebuggingConfiguration
+    * later to be created: SecureServingConfiguration, AuthenticationConfiguration, AuthorizationConfiguration, etc.
+* Assignee: ?
+
+#### kubelet changes
+
+* **GroupVersionKind:** `kubelet.config.k8s.io/v1beta.KubeletConfiguration`
+* **External Package:** `k8s.io/kubelet/config/v1beta1/types.go`
+* **Internal Package:** `k8s.io/kubernetes/pkg/kubelet/apis/config/types.go`
+* **Internal Scheme:** `k8s.io/kubernetes/pkg/kubelet/apis/config/scheme/scheme.go`
+* **Conversions & defaulting Package:** `k8s.io/kubernetes/pkg/kubelet/apis/config/v1beta1/register.go`
+* **Future Internal Package:** `k8s.io/kubelet/pkg/apis/config/types.go`
+* Assignee: @mtaufen?
+
+#### kube-proxy changes
+
+* **GroupVersionKind**: `kubeproxy.config.k8s.io/v1alpha1.KubeProxyConfiguration`
+* **External Package**: `k8s.io/kube-proxy/config/v1alpha1/types.go`
+* **Internal Package**: `k8s.io/kubernetes/pkg/proxy/apis/config/types.go`
+* **Internal Scheme**: `k8s.io/kubernetes/pkg/proxy/apis/config/scheme/scheme.go`
+* **Conversions & defaulting Package:** `k8s.io/kubernetes/pkg/proxy/apis/config/v1alpha1/register.go`
+* **Future Internal Package:** `k8s.io/kube-proxy/pkg/apis/config/types.go`
+* Start referencing `ClientConnectionConfiguration` from the generic ComponentConfig packages
+* Assignee: ?
+
+#### kube-scheduler changes
+
+* **GroupVersionKind**: `kubescheduler.config.k8s.io/v1alpha1.KubeSchedulerConfiguration`
+* **External Package**: `k8s.io/kube-scheduler/config/v1alpha1/types.go`
+* **Internal Package**: `k8s.io/kubernetes/pkg/scheduler/apis/config/types.go`
+* **Internal Scheme**: `k8s.io/kubernetes/pkg/scheduler/apis/config/scheme/scheme.go`
+* **Conversions & defaulting Package:** `k8s.io/kubernetes/pkg/scheduler/apis/config/v1alpha1/register.go`
+* **Future Internal Package:** `k8s.io/kube-scheduler/pkg/apis/config/types.go`
+* Start referencing `ClientConnectionConfiguration` & `LeaderElectionConfiguration` from the generic ComponentConfig packages
+* Assignee: @dixudx
+
+#### k8s.io/controller-manager changes
+
+* **Not a "real" API group, instead shared packages only with both external and internal types.**
+* **External Package with defaulting & conversions**: `k8s.io/controller-manager/pkg/apis/config/v1alpha1/types.go`
```

Comment:
If we want to change the `default value` easy, can we put the `default.go` in core k8s directory? such `pkg/`. May be a immature suggestions